### PR TITLE
[WEB-5183] Creating the InteractivePurchase message

### PR DESCRIPTION
**Pull request by ben-farrelly**
This message captures:
* User ID
* Products
* Order Date
* Order ID

This data in the message will be used in the SUP app to update the user profile.

### CODE REVIEW

**Review by mjliang — COMMENTED** [review]
Looks good, I just have one question. Do we need more test cases to cover more scenarios? For example `Make an order with the wrong format of the data`

**Comment by ben-farrelly** [comment]
> Looks good, I just have one question. Do we need more test cases to cover more scenarios? For example `Make an order with the wrong format of the data`

Hmm I was thinking about this, but this is type safe so the only thing to test is the product array, but I think that's better suited to the SUP app rather than the SDK, the SDK just wants some product data and the app has the schema for the product information.
Typically it seems like we don't do that schema validation at the SDK level. 
Although it's an interesting thought, what do you think?

File: src/Message/InteractivePurchase.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+     * @param array $params      Array of message parameters
+     * @return self
+     */
+    public static function create(int $userId, array $params = []): self
+    {
+        return new static($userId, $params);
+    }
+
+    /**
+     * Set the Product ID of the interactive purchase
```
Comment (takayuki-serato) [inline src/Message/InteractivePurchase.php:40]: Documentation is wrong G and can you chuck the schema for the array on here too pls

File: src/Message/InteractivePurchase.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+     * @return self
+     */
+    public function setProducts(array $products): self
+    {
+        $this->setParam(self::PRODUCTS, $products);
+        return $this;
+    }
+
+    /**
+     * Get the Product ID from an interactive purchase
```
Comment (takayuki-serato) [inline src/Message/InteractivePurchase.php:52]: No

**Review by takayuki-serato — CHANGES_REQUESTED** [review]
It was fine until I noticed you used copy and paste

File: src/Message/InteractivePurchase.php (hunk 2 above)
Comment (ben-farrelly) [reply src/Message/InteractivePurchase.php:52]: delet

**Review by takayuki-serato — APPROVED** [review]

**Comment by ben-farrelly** [comment]
Are you okay with this PR to be merged @mjliang ?

**Review by matiaslg13 — APPROVED** [review]
I checked the PR and also MJ's suggestions. I think, like Ben said, those checks should be at a SUP level, not SDK.

**Comment by ben-farrelly** [comment]
Okay, I'm going to merge this now and get this published to packagist.

**Pull request merged**